targets/TARGET_NORDIC/TARGET_NRF5x/TARGET_NRF51/sleep.c

```diff
@@ -59,21 +46,10 @@ void hal_sleep(void)
         // by using SEV/WFE pair, and then execute WFE again, unless there is
         // a pending interrupt.
 
-        // Set an event and wake up whatsoever, this will clear the event
-        // register from all previous events set (SVC call included)
+        __DSB();
         __SEV();
         __WFE();
-
-        // Test if there is an interrupt pending (mask reserved regions)
-        if (SCB->ICSR & (SCB_ICSR_RESERVED_BITS_MASK)) {
-            // Ok, there is an interrut pending, no need to go to sleep
-            return;
-        } else {
-            // next event will wakeup the CPU
-            // If an interrupt occured between the test of SCB->ICSR and this
-            // instruction, WFE will just not put the CPU to sleep
-            __WFE();
-        }
+        __WFE();
```

Comment:
What happens to interrupts that became pending before the first call to `__WFE` ?